File: 0038-count-and-say/0038-count-and-say.go
func countAndSay(n int) string {
	str := "1"

	for i := 2; i <= n; i++ {
		newStr := ""
		l := 0
		for r := 0; r < len(str); {
			for r < len(str) && str[l] == str[r] {
				r++
			}
			newStr += string(byte(r - l + '0'))
			newStr += string(str[l])
			l = r
		}
		str = newStr
	}
	return str
}

File: 0038-count-and-say/0038-count-and-say.cpp
class Solution {
   public:
    string countAndSay(int n) {
        string str = "1";
        for (int i = 2; i <= n; i++) {
            string newStr;
            int l = 0;
            for (int r = 0; r < str.size();) {
                while (r < str.size() && str[l] == str[r]) {
                    r++;
                }
                newStr += char(r - l + '0');
                newStr += str[l];
                l = r;
            }
            str = newStr;
        }

        return str;
    }
};